sdk/core/Azure.Core.TestFramework/src/TestDiagnosticListener.cs

```diff
@@ -7,6 +7,7 @@
 
 namespace Azure.Core.Tests
 {
+    // DO NOT USE - use ClientDiagnosticListener instead
```

Comment:
Should we consider marking as Obsolete?

Reply:
Yes!

Reply:
There is still code using it so it would cause the build to fail.

Reply:
I can just update those tests and delete this file entirely. @pakrym does that work?

Reply:
I have the change in another long-running PR. You can suppress the Obsolete warning at the call site.

Reply:
what change?

Reply:
Ah nevermind - I will just update the callsites to ignore the warning - I see that it would not be as simple as just swapping out the types.

Reply:
>"I can just update those tests"

I thought I flipped the activity policy tests to diagnostic scope listener as part of https://github.com/Azure/azure-sdk-for-net/pull/16371/files

Seems like I didn't. I'll be OK with you making the change and removing the file.

Reply:
There were many more instances than what appeared in GitHub search.. I disabled the warnings for now.

Reply:
For some reason the build was failing even after suppressing the warnings at the call sites.